### PR TITLE
remove remote simulator

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Remove all the remote simulators

# Related issues

Solves https://github.com/CQCL/pytket-qiskit/issues/300 https://github.com/CQCL/pytket-qiskit/issues/278 https://github.com/CQCL/pytket-qiskit/issues/272 https://github.com/CQCL/pytket-qiskit/issues/231 https://github.com/CQCL/pytket-qiskit/pull/287

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: docs/intro.txt [hunk 1]
```diff
@@ -95,7 +94,7 @@ to disk:
     IBMProvider.save_account(token=ibm_token)
     QiskitRuntimeService.save_account(channel="ibm_quantum", token=ibm_token)
 
-To see which devices you can access you can use the ``available_devices`` method on the :py:class:`IBMQBackend` or :py:class:`IBMQEmulatorBackend`. Note that it is possible to pass optional ``instance`` and ``provider`` arguments to this method. This allows you to see which devices are accessible through your IBM hub.
```
Comment (cqc-melf) [inline docs/intro.txt:98]: The local emulator was not listed here? Is this functionality available with the local setup?

**Comment by cqc-melf** [comment]
I have tried to slip up the changes into different commits, looking at them separately might be helpful in the review.

File: docs/intro.txt (hunk 1 above)
Comment (cqc-alec) [reply docs/intro.txt:98]: > The local emulator was not listed here? Is this functionality available with the local setup?

No, it's just a class method on `IBMQBackend`.

File: docs/changelog.rst [hunk 2]
```diff
@@ -5,6 +5,8 @@ Unreleased
 ----------
 
 * Update qiskit-ibm-runtime version requirement to 0.22.
+* remove all remote simulators
+* rename ``IBMQLocalEmulatorBackend`` to ``IBMQEmulatorBackend``
```
Comment (CalMacCQ) [inline docs/changelog.rst:9]: Maybe add that simulatons that used `IBMQEmulatorBackend` previously will now run locally. Just to be explcit.

File: tests/backend_test.py [hunk 3]
```diff
@@ -836,12 +834,11 @@ def test_operator_expectation_value() -> None:
...
 @pytest.mark.skipif(skip_remote_tests, reason=REASON)
 def test_ibmq_emulator(
-    brisbane_emulator_backend: IBMQEmulatorBackend,
-    brisbane_local_emulator_backend: IBMQLocalEmulatorBackend,
+    brisbane_local_emulator_backend: IBMQEmulatorBackend,
 ) -> None:
-    for b in [brisbane_emulator_backend, brisbane_local_emulator_backend]:
-        assert b._noise_model is not None  # type: ignore
-        b_ibm = b._ibmq  # type: ignore
+    for b in [brisbane_local_emulator_backend]:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply tests/backend_test.py]: Done in [bff3e00](https://github.com/CQCL/pytket-qiskit/pull/301/commits/bff3e001904cfcc03a338df13b1863b599b47d8a)

File: tests/backend_test.py [hunk 4]
```diff
@@ -1250,11 +1190,10 @@ def test_sim_qubit_order() -> None:
 @pytest.mark.flaky(reruns=3, reruns_delay=10)
 @pytest.mark.skipif(skip_remote_tests, reason=REASON)
 def test_required_predicates(
-    brisbane_emulator_backend: IBMQEmulatorBackend,
-    brisbane_local_emulator_backend: IBMQLocalEmulatorBackend,
+    brisbane_local_emulator_backend: IBMQEmulatorBackend,
 ) -> None:
     # https://github.com/CQCL/pytket-qiskit/issues/93
-    for b in [brisbane_emulator_backend, brisbane_local_emulator_backend]:
+    for b in [brisbane_local_emulator_backend]:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply tests/backend_test.py]: Done in [bff3e00](https://github.com/CQCL/pytket-qiskit/pull/301/commits/bff3e001904cfcc03a338df13b1863b599b47d8a)

File: tests/backend_test.py [hunk 5]
```diff
@@ -1113,62 +1110,6 @@ def test_postprocess() -> None:
     b.cancel(h)
 
 
-@pytest.mark.flaky(reruns=3, reruns_delay=10)
-@pytest.mark.skipif(skip_remote_tests, reason=REASON)
-def test_postprocess_emu(ibmq_qasm_emulator_backend: IBMQEmulatorBackend) -> None:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply tests/backend_test.py:1118]: I have added an issue for that https://github.com/CQCL/pytket-qiskit/issues/302 

I will try to do this later today, would you be fine with doing a release without this?

File: docs/changelog.rst (hunk 2 above)
Comment (cqc-melf) [reply docs/changelog.rst:9]: done in [a7a2470](https://github.com/CQCL/pytket-qiskit/pull/301/commits/a7a247060088f8eb09a6785f2af3659df9f73f1c)

File: tests/conftest.py [hunk 6]
```diff
@@ -59,45 +58,8 @@ def qasm_simulator_backend() -> IBMQBackend:
...
-    return IBMQEmulatorBackend(
-        "ibm_brisbane",
-        instance="ibm-q/open/main",
-        token=os.getenv("PYTKET_REMOTE_QISKIT_TOKEN"),
-    )
-
-
-@pytest.fixture(scope="module")
-def ibmq_qasm_emulator_backend() -> IBMQEmulatorBackend:
+def brisbane_local_emulator_backend() -> IBMQEmulatorBackend:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply tests/conftest.py]: Done in [b17a67e](https://github.com/CQCL/pytket-qiskit/pull/301/commits/b17a67e5b4e0ddb0835a92748f24e19c9f114887)

File: pytket/extensions/qiskit/backends/ibmq_local_emulator.py
[the remark this answers is not in the document]
Comment (cqc-melf) [reply pytket/extensions/qiskit/backends/ibmq_local_emulator.py:1]: I have done this now in a separate commit, if you don't try to review all changes at once it should be fine.

Done in [12eb951](https://github.com/CQCL/pytket-qiskit/pull/301/commits/12eb951d3682b668dac86a3b44f5cfd6a13126e2)

**Review by cqc-alec — APPROVED** [review]

**Review by CalMacCQ — APPROVED** [review]

**Pull request merged**